pkg/odo/cli/component/push.go

```diff
@@ -51,6 +51,14 @@ var pushCmd = &cobra.Command{
 		}
 		componentName := context.Component(argComponent)
 
+		// if the componentName is blank then there is no active component set
+		if len(componentName) == 0 {
+			log.Error("No component is set as active.\n" +
+				"Use 'odo component set' to set and existing component as active \n" +
+				"or call this command with component name as an argument.")
```

Comment:
This is too long, I'd always try to avoid doing multi-line error messages. Just simply log error `No component is set as active. Use 'odo component set' to set an active component`

Reply:
Yeah, I'd avoid multi-line as that messes up any other output :+1: Highly suggest a one-liner :)

Reply:
ack and resolved